homeassistant/components/sensor/ebusd.py

```diff
@@ -0,0 +1,251 @@
+                _LOGGER.warning("Element not found: %s", name)
+                raise RuntimeError("Element not found")
+            else:
+                self.value[name] = command_result
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+            raise RuntimeError("socket timeout")
+        except socket.error:
+            _LOGGER.error("socket error: %s", socket.error)
+            raise RuntimeError("Command failed")
+
+    def write(self, call):
+        """Call write methon on ebusd."""
+        import ebusdpy
+        name = call.data.get('name')
+        value = call.data.get('value')
+        command = WRITE_COMMAND.format(self._circuit, name, value)
+
+        try:
+            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
+            command_result = ebusdpy.send_command(self._address, command)
+            if 'done' not in command_result:
+                _LOGGER.warning('Write command failed: %s', name)
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+        except socket.error:
+            _LOGGER.error()
+
+
+class Ebusd(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, data, sensor_type, name):
+        """Initialize the sensor."""
+        self._state = None
+        self._client_name = name
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self._type = SENSOR_TYPES[sensor_type][3]
+        self.data = data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self._client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Fetch new state data for the sensor."""
+        try:
+            self.data.update(self._name)
+            if self._name in self.data.value:
+                if self._type == 0:
+                    self._state = format(
+                        float(self.data.value[self._name]), '.1f')
+                elif self._type == 1:
+                    self._state = timer_format(self.data.value[self._name])
+                elif self._type == 2:
+                    if self.data.value[self._name] == 1:
+                        self._state = STATE_ON
+                    else:
+                        self._state = STATE_OFF
+                elif self._type == 3:
+                    self._state = self.data.value[self._name]
+        except RuntimeError:
+            _LOGGER.debug("EbusdData.update exception")
```

Comment:
Set `available` property to False when this happens.

Reply:
That will be an issue cause most of the time ebusd returns an error so the sensor value will blink between not available and the correct value. In my opinion it is better to have the previous stored value instead of nothing.

Reply:
I agree with CrazYoshi, Ebus is a serial interface with signal carried usually over two straight (not screened, not twisted) cables.  Theoretically random interferences can appear, but shouldn't disrupt the functionality.